junit-jupiter-params/src/main/java/org/junit/jupiter/params/support/AnnotationConsumerInitializer.java

```diff
@@ -47,14 +52,28 @@ private AnnotationConsumerInitializer() {
 	public static <T> T initialize(AnnotatedElement annotatedElement, T annotationConsumerInstance) {
 		if (annotationConsumerInstance instanceof AnnotationConsumer) {
 			Class<? extends Annotation> annotationType = findConsumedAnnotationType(annotationConsumerInstance);
-			Annotation annotation = AnnotationUtils.findAnnotation(annotatedElement, annotationType) //
-					.orElseThrow(() -> new JUnitException(annotationConsumerInstance.getClass().getName()
-							+ " must be used with an annotation of type " + annotationType.getName()));
-			initializeAnnotationConsumer((AnnotationConsumer) annotationConsumerInstance, annotation);
+			List<? extends Annotation> annotations = streamAnnotations(annotatedElement, annotationType);
+
+			if (annotations.isEmpty()) {
+				throw new JUnitException(annotationConsumerInstance.getClass().getName()
+						+ " must be used with an annotation of type " + annotationType.getName());
+			}
+
+			annotations.forEach(annotation -> initializeAnnotationConsumer(
+				(AnnotationConsumer) annotationConsumerInstance, annotation));
 		}
 		return annotationConsumerInstance;
 	}
 
+	private static List<? extends Annotation> streamAnnotations(AnnotatedElement annotatedElement,
+			Class<? extends Annotation> annotationType) {
```

Comment:
```suggestion
	private static <T extends Annotation> List<T> findAnnotations(AnnotatedElement annotatedElement,
			Class<T> annotationType) {
```

Reply:
Solved in f1046d205ceab97f58aa656818dd0fa6099d7277